server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -164,27 +182,22 @@ private MergeResult partialReduce(MergeTask task,
             newTopDocs = null;
         }
 
-        final DelayableWriteable.Serialized<InternalAggregations> newAggs;
+        final InternalAggregations newAggs;
         if (hasAggs) {
             List<InternalAggregations> aggsList = new ArrayList<>();
             if (lastMerge != null) {
-                aggsList.add(lastMerge.reducedAggs.expand());
+                aggsList.add(lastMerge.reducedAggs);
             }
             for (QuerySearchResult result : toConsume) {
                 aggsList.add(result.consumeAggs().expand());
             }
             InternalAggregations result = InternalAggregations.topLevelReduce(aggsList,
                 aggReduceContextBuilder.forPartialReduction());
-            newAggs = DelayableWriteable.referencing(result).asSerialized(InternalAggregations::readFrom, namedWriteableRegistry);
-            long previousBufferSize = aggsCurrentBufferSize;
-            aggsCurrentBufferSize = newAggs.ramBytesUsed();
-            aggsMaxBufferSize = Math.max(aggsCurrentBufferSize, aggsMaxBufferSize);
-            logger.trace("aggs partial reduction [{}->{}] max [{}]",
-                previousBufferSize, aggsCurrentBufferSize, aggsMaxBufferSize);
+            newAggs = result;
```

Comment:
I think you can skip the `result` variable now.

Reply:
++, https://github.com/elastic/elasticsearch/pull/62223/commits/7178dab4566971b0b0daee19f2b22a8f597f61b1